src/group_theory/abelianization.lean

```diff
@@ -83,6 +83,7 @@ end lift
 
 variables {A : Type v} [monoid A]
 
+@[ext, priority 1500]
```

Comment:
We don't have this priority in `free_algebra` etc, yet it seems to already do the right thing. Any idea why?

Reply:
Nope. I just copied this priority from some other `hom_ext` that I found. I guess someone who actually knows what the right thing is can help us by writing a library note.

Reply:
Here's the one I'm thinking of:

https://github.com/leanprover-community/mathlib/blob/61e76c42054a081ee5e8d58500eab948c3c3ac33/src/algebra/free_algebra.lean#L291-L297

Reply:
There's a comment here that could be lifted to a library note:

https://github.com/leanprover-community/mathlib/blob/e5ea2000f5781e7070055b067d5e688a61853354/src/algebra/monoid_algebra.lean#L338-L339

Reply:
And a few more variants of it here: https://github.com/leanprover-community/mathlib/search?q=type-specific+extensionality+lemmas